crates/iota/src/client_commands.rs

```diff
@@ -3047,3 +3051,35 @@ fn parse_emit_option(s: &str) -> Result<HashSet<EmitOption>, String> {
 
     Ok(options)
 }
+
+/// Warn the user if the CLI does not match the version of current on-chain
+/// protocol.
+async fn check_protocol_version_and_warn(client: &IotaClient) -> Result<(), anyhow::Error> {
+    let on_chain_protocol_version = client
+        .read_api()
+        .get_protocol_config(None)
+        .await?
+        .protocol_version
+        .as_u64();
+    let cli_protocol_version = ProtocolVersion::MAX.as_u64();
+
+    if cli_protocol_version != on_chain_protocol_version {
+        let warning_msg = format!(
+            "[warning] The CLI's protocol version is {cli_protocol_version}, but the active \
+            network's protocol version is {on_chain_protocol_version}."
+        );
+        let help_msg = if cli_protocol_version < on_chain_protocol_version {
+            "Consider installing the latest version of the CLI - \
+            https://docs.iota.org/references/cli \n\n \
+            If publishing/upgrading returns a dependency verification error, then install the \
+            latest CLI version."
+        } else {
+            "Consider waiting for the network to have upgraded to the same version, \
+            or using the previous version of the CLI for this operation."
```

Comment:
```suggestion
            or using a previous version of the CLI for this operation."
```
"the" previous version could result with the exact same issue I guess?

Reply:
hmm, but "a" previous version could be any prior version, which would result in the other warning for it being out of date :troll: . I think `the` is fine, because it's fair to assume, that the network gets updated with the new version, before a new CLI is released.